src/bandicoot.js

```diff
@@ -9,3 +9,5 @@ export {useLink} from './use-link.hook.js'
 export {useContentEditableFalse} from './use-content-editable-false.hook.js'
 export {useTextAsImage} from './use-text-as-image.hook.js'
 export {useElementDeletionDetection} from './use-element-deletion-detection.hook.js'
+
+export { BoldButton } from './icon-buttons/bold-button.component'
```

Comment:
As indicated [in my comment](https://github.com/CanopyTax/bandicoot/issues/60), I think I'd prefer exporting BoldIcon and IconButton **separately**.

That way we don't have to make these components support passing in custom click listeners or css classNames or other props like that. The user of bandicoot will still be required to use the bandicoot hooks and IconButton/BoldIcon components themselves, but they just won't have to worry about finding a default icon set or hover state.

Reply:
What custom click listeners need to be provided (fileupload perhaps)? I would hope those would be the exception. As far as className, I just don't see the problem as big enough to task the developer with so much. My bias is toward the simple devex

```jsx
<RichTextContainer>
  <div>
    <BoldButton />
    <ItalicButton />
    <UnderlineButton />
  </div>
  <RichTextEditor />
</RichTextContainer>
```

Reply:
```jsx
    <button
      {...props}
      className={`
        ${style.bandicootButton}
        ${props.isActive ? 'active-control-button' : ''}
        ${props.className || ''}
      `}
```
In which case will classname be desired and this application to the button suboptimal?

Reply:
there could be need for styling both the icon and the button separately. Your implementation does not allow the user to apply a classname to the button, which is a non-starter for this in my opinion. Many users will want to style both the button and the icon however they want.

I agree that what you've proposed a simpler DX, but it can't do what is needed and also it's harder to maintain the bandicoot code in the long term.

Reply:
I'm still failing to understand the benefits of not abstracting.

> Your implementation does not allow the user to apply a classname to the button

I believe it does. see `${props.className || ''}` on button in my previous comment

> there could be need for styling both the icon and the button separately

Sure, but that's not too difficult to do if needed. My proposal:

```jsx
const styles = `
  .abc { /* button styles */ }
  .abc svg { /* icon styles */ }
`
<BoldButton className='abc'/>
```

Alternatively, 
```jsx
<IconButton
      onClick={performCommand}
      className={`${isActive ? 'active-control-button' : ''} abc`}
    >
      <BoldIcon className='abcSVG' />
    </button>
```
Am I not seeing how this could be approached better?

> Many users will want to style both the button and the icon however they want.

That is very possible. How can we provide default styling (should there be any default styling like hover?) but then easily toggle/override?

The one disadvantage to my proposal in the styling is we have to use a default style for `active-control-button` but that could be resolved with an additional prop or just good documentation. I actually like the idea of having this class that gets applied and then letting the dev apply desired style to that class.

> it's harder to maintain the bandicoot code in the long term.

I lack the experience to agree or disagree but I think it's worth bearing the burden for better DX. The only question I'd have is will this lead to more breaking changes than desired?

Reply:
I don’t like your proposal for a shared className prop for two components, for the following reasons:

1. Many react css libraries do not allow you to create arbitrary css selectors for svg’s and buttons that are children of your css class. See styled components and emotion as examples of where it’s not easy to do that
2. It is weird that the css selector would have to be `button.foo` for the button, but `.foo svg` for the svg. Would not be obvious to user that they need to do that
3. Having the user of bandicoot select button and svg elements rendered by bandicoot locks bandicoot into a specific implementation. If we were to switch to new icons that don’t have svgs, for example, we wouldn’t be able to do that without a breaking change.
4. What you’ve proposed is not a normal pattern and would require documentation and explanation. The abstraction is not immediately understood without reading about it

Reply:
1. Forgive my ignorance. I don't know enough about those CSS libraries.
```jsx
import {IconButton, BoldIcon} from 'bandicoot'
const Button = styled(IconButton)`
  /* button styling */
`
const Icon = styled(BoldIcon)`
  /* icon styling */
`
```
```jsx
import { BoldButton } from 'bandicoot'
const Button = styled(BoldButton)`
  /* there is probably a hacky way to get to the child (svg) but probably not pretty I think you're right */
`
```
2. (and 4) No less obvious than importing both the button and icon as separate components imo. Both will take documentation for explanation.
3. if we switch icons to something other than SVG or introduce new elements this will indeed introduce a breaking change but how likely is that?

I will move toward the two component export because while I would slightly prefer the single component, you are the primary architect here and I yield to your expertise.

Question, what should we do if there are more than one children passed to button?
```jsx
<IconButton
    onClick={performCommand}
    className={`${isActive ? 'active-control-button' : ''} abc`}
  >
    <div>hey why are you here?</div>
    <BoldIcon className='abcSVG' />
    <ARandomComponent />
  </button>
```

I'm thinking check for children count and check to see if child is of expected type (Icon). Not sure the best way to do this in react but will figure it out

Reply:
I honestly don't have much desire to continue debating this with you. Since you're the one implementing it, I'm happy to go with what you think is best.